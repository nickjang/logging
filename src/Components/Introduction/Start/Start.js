import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';

import 'ProjectSelect.css';

class ProjectSelect extends Component {
  handleSignUpClick = (e) => {
    e.preventDefault();
    this.props.history.push('/sign-up');
  }

  handleDemoClick = (e) => {
    // Log into demo account
    e.preventDefault();
    this.props.history.push('/');
  }

  render() {
    return (
      <main>
        <section>
          <p>Create an account, or use the demo account.</p>
          <p>With the demo account, you’ll be able to use the app’s features; however, logs made with the demo account won’t be saved to the server. You’ll also see demo logs already in your view page.</p>
          <button onClick={(e) => { this.handleSignUpClick(e) }}>Sign Up</button>
          <button onClick={(e) => { this.handleDemoClick(e) }}>Demo</button>
        </section>
      </main>
    );
  }
}

export default withRouter(ProjectSelect);