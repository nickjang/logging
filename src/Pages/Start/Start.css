.start h2 {
  text-align: center;
}

.start {
  display: flex;
  flex-direction: column;
  width: fit-content;
  margin: 2rem auto;
  padding: 0 1rem;
  align-items: center;
}

.start span{
  display: block;
}