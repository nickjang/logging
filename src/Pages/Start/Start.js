import React, { Component } from 'react';
import './Start.css';

class Start extends Component {
  handleSignUpClick = (e) => {
    e.preventDefault();
    this.props.history.push('/sign-up');
  }

  handleDemoClick = (e) => {
    // Log into demo account
    e.preventDefault();
    this.props.history.push('/overview');
  }

  render() {    
    return (
      <div>
        <section>
          <p>Create an account, or use the demo account.</p>
          <p>With the demo account, you’ll be able to use the app’s features; however, logs made with the demo account won’t be saved to the server. You’ll also see demo logs already in your view page.</p>
          <button onClick={(e) => { this.handleSignUpClick(e) }}>Sign Up</button>
          <button onClick={(e) => { this.handleDemoClick(e) }}>Demo</button>
        </section>
      </div>
    );
  }
}

export default Start;