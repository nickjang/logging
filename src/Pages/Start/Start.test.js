import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Start from './Start';

describe(`Start component`, () => {
  it('renders Start by default', () => {
    const wrapper = shallow(<Start />);
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})