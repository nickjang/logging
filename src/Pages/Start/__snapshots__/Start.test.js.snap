// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start component renders Start by default 1`] = `
<section
  className="start"
>
  <h2
    className="lg-title"
  >
    Create an account, or use the demo account.
  </h2>
  <p>
    With the demo account, youâ€™ll be able to use all the apps features using a shared public account.
  </p>
  <span
    className="status "
  />
  <div>
    <button
      className="lg-btn lg-btn-light mt-2 mr-1"
      onClick={[Function]}
    >
       Sign Up
    </button>
    <button
      className="lg-btn lg-btn-light mt-2"
      onClick={[Function]}
    >
       Demo
    </button>
  </div>
</section>
`;
